instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/HttpSpanAttributes.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awslambda.v1_0;
+
+import static io.opentelemetry.api.trace.attributes.SemanticAttributes.HTTP_METHOD;
+import static io.opentelemetry.api.trace.attributes.SemanticAttributes.HTTP_STATUS_CODE;
+import static io.opentelemetry.api.trace.attributes.SemanticAttributes.HTTP_URL;
+import static io.opentelemetry.api.trace.attributes.SemanticAttributes.HTTP_USER_AGENT;
+import static io.opentelemetry.instrumentation.awslambda.v1_0.MapUtils.emptyIfNull;
+import static io.opentelemetry.instrumentation.awslambda.v1_0.MapUtils.lowercaseMap;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+final class HttpSpanAttributes {
```

Comment:
Do you think it makes sense to have a `AwsLambdaGatewayProxyTracer extends HttpServerTracer`?

Reply:
I'd considered choosing different tracers based on the input class, but `AwsLambdaTracer` has some AWS specific logic (context propagation, `faas.*` attributes) that I'd need to duplicate in a new tracer (which would extend `HttpServerTracer`); I'd have to put the AWS logic in a separate helper class if I were to do that.
So out of the two choices: AWS logic in tracer vs HTTP logic in tracer I'd decided to put the HTTP things in the second place; it seemed to me that it'd be easier to add support for other event types this way (share the same tracer class with all the AWS things, separate strategy for additional attributes)